src/main/java/me/jellysquid/mods/sodium/client/util/workarounds/InGameChecks.java

```diff
@@ -0,0 +1,109 @@
+            "fog.glsl"
+    ));
+
+    /**
+     * <a href="https://github.com/CaffeineMC/sodium-fabric/issues/1569">#1569</a>
+     * Iterate through all active resource packs, and detect resource packs which contain files matching the blacklist.
+     * An error message is shown for resource packs which replace terrain core shaders.
+     * A warning is shown for resource packs which replace the default light.glsl and fog.glsl shaders.
+     * Detailed information on shader files replaced by resource packs is printed in the client log.
+     */
+    public static void checkIfCoreShaderLoaded(ResourceManager manager) {
+        HashMap<String, MessageLevel> detectedResourcePacks = new HashMap<>();
+        var customResourcePacks = manager.streamResourcePacks();
+
+        customResourcePacks.forEach(resourcePack -> {
+            // Omit 'vanilla' and 'fabric' resource packs
+            if (!resourcePack.getName().equals("vanilla") && !resourcePack.getName().equals("fabric")) {
+                var resourcePackName = resourcePack.getName();
+
+                resourcePack.findResources(ResourceType.CLIENT_RESOURCES, Identifier.DEFAULT_NAMESPACE, "shaders", (path, ignored) -> {
+                    // Trim full shader file path to only contain the filename
+                    var shaderName = path.getPath().substring(path.getPath().lastIndexOf('/') + 1);
+                    if (vshBlacklist.contains(shaderName)) {
+
+                        if (!detectedResourcePacks.containsKey(resourcePackName)) {
+                            detectedResourcePacks.put(resourcePackName, MessageLevel.SEVERE);
+                        } else if (detectedResourcePacks.get(resourcePackName) == MessageLevel.WARN) {
+                            detectedResourcePacks.replace(resourcePackName, MessageLevel.SEVERE);
+                        }
+
+                        LOGGER.error("Resource pack '" + resourcePackName + "' replaces core shader '" + shaderName + "'");
+                    }
+
+                    if (glslBlacklist.contains(shaderName)) {
+
+                        if (!detectedResourcePacks.containsKey(resourcePackName)) {
+                            detectedResourcePacks.put(resourcePackName, MessageLevel.WARN);
+                        }
+
+                        LOGGER.warn("Resource pack '" + resourcePackName + "' replaces shader '" + shaderName + "'");
+
+                    }
+                });
+            }
+        });
+
+        if (detectedResourcePacks.containsValue(MessageLevel.SEVERE)) {
+            showConsoleMessage(Text.translatable("sodium.console.core_shaders_error"), MessageLevel.SEVERE);
+
+            for (Map.Entry<String, MessageLevel> entry : detectedResourcePacks.entrySet()) {
+
+                if (entry.getValue() == MessageLevel.SEVERE) {
+                    // Omit 'file/' prefix for the in-game message
+                    showConsoleMessage(Text.literal(entry.getKey().substring(5)), MessageLevel.SEVERE);
```

Comment:
Since built-in resourcepacks are also scanned, wouldn't their name also be affected by this code?